src/hooks/useApproveVault.ts

```diff
@@ -0,0 +1,96 @@
+import { BigNumber, Contract, ethers } from 'ethers';
+import { useCallback, useMemo, useState, useEffect } from 'react';
+
+import useDanteFinance from './useDanteFinance';
+
+import useInterval from './useInterval';
+
+import ERC20 from '../dante-finance/ERC20';
+
+const APPROVE_AMOUNT = ethers.constants.MaxUint256;
+const APPROVE_BASE_AMOUNT = BigNumber.from('1000000000000000000000000');
+
+export enum ApprovalState {
+  UNKNOWN,
+  NOT_APPROVED,
+  PENDING,
+  APPROVED,
+}
+
+// returns a variable indicating the state of the approval and a function which approves if necessary or early returns
+function useApproveVault(
+  lp: string,
+  vault: string,
+): [ApprovalState, () => Promise<void>] {
+  const danteFinance = useDanteFinance();
+
+  const token: ERC20 = danteFinance.externalTokens[lp];
+  const router: Contract = danteFinance.contracts[vault];
+
+  const [pending, setPending] = useState(false);
+  const [currentAllowance, setAllowance] = useState<BigNumber>(
```

Comment:
if the state is called `currentAllowance ` the setter should be named `setCurrentAllowance`